### PR TITLE
 Enable live pprof at /.pprof/

### PR DESCRIPTION
**Pull request by vadmeste**
Running `go tool pprof
http://localhost:8080/minio/pprof/profile?seconds=30` will get CPU
profiling for 20 seconds

Supported profilers are: goroutine, threadcreate, heap, allocs, block,
mutex, symbol, trace and profile (cpu profiling)

### CODE REVIEW

**Comment by klauspost** [comment]
@vadmeste I find the HTTP profilers to be much more useful since it doesn't require restarting. Maybe we could expose them on a separate port, so adding a parameter `--pprof=":6060"` would expose profiling on a separate port. 

> There is also a standard HTTP interface to profiling data. In an HTTP server, adding
> `import _ "net/http/pprof"` will install handlers for a few URLs under /debug/pprof/. Then you can run go tool pprof with a single argument—the URL to your server’s profiling data and it will download and examine a live profile.

```
go tool pprof http://localhost:6060/debug/pprof/profile   # 30-second CPU profile
go tool pprof http://localhost:6060/debug/pprof/heap      # heap profile
go tool pprof http://localhost:6060/debug/pprof/block     # goroutine blocking profile
```

- You will need to add the pprof to a custom mux, but there are easy functions for that. I can send a PR if you'd like.

**Comment by harshavardhana** [comment]
> @vadmeste I find the HTTP profilers to be much more useful since it doesn't require restarting. Maybe we could expose them on a separate port, so adding a parameter `--pprof=":6060"` would expose profiling on a separate port.
> 
> > There is also a standard HTTP interface to profiling data. In an HTTP server, adding
> > `import _ "net/http/pprof"` will install handlers for a few URLs under /debug/pprof/. Then you can run go tool pprof with a single argument—the URL to your server’s profiling data and it will download and examine a live profile.
> 
> ```
> go tool pprof http://localhost:6060/debug/pprof/profile   # 30-second CPU profile
> go tool pprof http://localhost:6060/debug/pprof/heap      # heap profile
> go tool pprof http://localhost:6060/debug/pprof/block     # goroutine blocking profile
> ```
> 
> * You will need to add the pprof to a custom mux, but there are easy functions for that. I can send a PR if you'd like.

This is better as @klauspost suggested

**Comment by vadmeste** [comment]
@klauspost I made the change, by the way, this does not require any flag.. it can be always enabled

**Comment by klauspost** [comment]
@vadmeste The reason for the separate port is for safety reasons. While adding it to the default port is a lot easier, both for use and setup it does allow anyone to access potentially resource intensive calls.

**Comment by klauspost** [comment]
Yes, why I wrote that "adding it to the default port is a lot easier, both for use and setup". But it doesn't look like something we would like to have exposed externally.

But it is easier to block a port.

File: main.go [hunk 1]
```diff
@@ -68,6 +69,10 @@ var (
 )
 
 const (
+	reservedPath  = "/minio"
+	profilingPath = reservedPath + "/pprof"
+
+	// TODO: replace /.prometheus with /minio/prometheus/metrics
```
Comment (harshavardhana) [inline main.go:75]: This is chosen on purpose since there can never be buckets starting with a `.`  - so you can use `.pprof`

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: main.go [hunk 2]
```diff
@@ -811,6 +817,10 @@ func main() {
 			Name:  "key",
 			Usage: "server private key file",
 		},
+		cli.StringFlag{
+			Name:  "pprof",
+			Usage: "start and and listen on the specified address (e.g. `:6060`)",
```
Comment (harshavardhana) [inline main.go:822]:
```suggestion
			Usage: "start and listen for profiling on the specified address (e.g. `:1337`)",
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**